pkg/operator/controller/instance_controller.go

```diff
@@ -0,0 +1,742 @@
+	}
+
+	// Set the deployment namespace
+	r.clNamespace = InstanceNamespace
+	if r.instance.Namespace != OperatorNameSpace { // Allow to create in any namespace. TODO -should remove this part and use only InstanceNamespace
+		r.clNamespace = r.instance.Namespace
+	}
+
+	// Examine DeletionTimestamp to determine if object is under deletion
+	if !r.instance.DeletionTimestamp.IsZero() {
+		if err := r.deleteAllResources(ctx); err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if err := r.deleteFinalizer(ctx); err != nil {
+			return ctrl.Result{}, err
+		}
+	}
+	// Set Finalizer if needed
+	if err := r.setFinalizer(ctx); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Set ClusterLink components if needed
+	if err := r.setClusterLink(ctx); err != nil {
+		return ctrl.Result{}, err
+	}
+	// Check ClusterLink components status
+	if err := r.checkStatus(ctx); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Wait until status is ready or error
+	if r.instance.Status.Controlplane.Conditions[string(api.DeploymentReady)].Reason == StatusModeProgressing ||
+		r.instance.Status.Dataplane.Conditions[string(api.DeploymentReady)].Reason == StatusModeProgressing {
+		return ctrl.Result{Requeue: true, RequeueAfter: time.Second * 5}, err
+	}
+	return ctrl.Result{}, nil
+}
+
+// setClusterLink sets up all the components for the ClusterLink project.
+func (r *InstanceReconciler) setClusterLink(ctx context.Context) error {
+
+	// Set Namespace
+	if err := r.setNameSpace(ctx); err != nil {
+		return err
+	}
+
+	// Set PVC
+	if err := r.setPVC(ctx, ControlPlaneName); err != nil {
+		return err
+	}
+
+	// Set services
+	if err := r.setService(ctx, ControlPlaneName, cpapi.ListenPort); err != nil {
+		return err
+	}
+	if err := r.setService(ctx, DataPlaneName, dpapi.ListenPort); err != nil {
+		return err
+	}
+
+	// Set deployments
+	if err := r.setControlplane(ctx); err != nil {
+		return err
+	}
+	if err := r.setDataplane(ctx); err != nil {
+		return err
+	}
+
+	// Set rules
+	if err := r.setRules(ctx, ControlPlaneName); err != nil {
+		return err
+	}
+
+	// Set External service
+	return r.setExternalService(ctx)
+}
+
+// setControlplane sets up the controlplane deployment.
+func (r *InstanceReconciler) setControlplane(ctx context.Context) error {
+	cpDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ControlPlaneName,
+			Namespace: r.clNamespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: int32Ptr(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": ControlPlaneName},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": ControlPlaneName},
+				},
+				Spec: corev1.PodSpec{
+					Volumes: []corev1.Volume{
+						{
+							Name: "ca",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: "cl-fabric",
+								},
+							},
+						},
+						{
+							Name: "tls",
```

Comment:
Is there a more descriptive name?
TLS of which component? Maybe certificates?

Reply:
This part of the controlplane- @orozery can change it in the new controlplane.